Modules/Scripted/VPAWModelOCT/VPAWModelOCT.py

```diff
@@ -0,0 +1,668 @@
+                        for _, install_name in needs_installation
+                        if not install_name.startswith("torch")
+                    ]
+                    with_torch = [
+                        install_name
+                        for _, install_name in needs_installation
+                        if install_name.startswith("torch")
+                    ]
+                    if without_torch:
+                        slicer.util.pip_install(without_torch)
+                    if with_torch:
+                        slicer.util.pip_install(
+                            [
+                                *with_torch,
+                                "--index-url",
+                                "https://download.pytorch.org/whl/cu117",
+                            ],
+                        )
+                    installed_modules = [
+                        (
+                            import_name,
+                            install_name,
+                            importlib.import_module(import_name),
+                        )
+                        for import_name, install_name in needs_installation
+                    ]
+            except ModuleNotFoundError as e2:
+                slicer.util.errorDisplay(
+                    "\n".join(
+                        [
+                            f"Unable to install package{plural}.",
+                            "Check the console for details.",
+                        ],
+                    ),
+                    "Install Error",
+                )
+                print(e2)
+                return False
+
+        # All dependencies were successfully imported
+        self.showInstalledModules(installed_modules)
+        return True
+
+    def runOCTSeg(
+        self,
+        octSegDirectory,
+        vPAWOCTConfigFile,
+    ):
+        """
+        Run the OCT Segmentation pipeline.
+        Can be used without GUI widget.
+
+        Parameters
+        ----------
+        octSegDirectory : str
+            This is the directory containing the source code for OCTSeg.
+        vPAWOCTConfigFile : str
+            This is the full path name to a top-level configuration file such as
+            uncbiag/OCTSeg/configs/unet_v2/unet_v2_test_1_channel.yaml.
+        """
+        # If self.OCTSeg_directory is not yet set then see if we can set it.
+        if not hasattr(self, "OCTSeg_directory") and not (
+            os.path.isdir(octSegDirectory) and self.linkOCTSeg(octSegDirectory)
+        ):
+            # We don't have self.OCTSeg_directory and we couldn't get it.
+            return False
+
+        startTime = time.time()
+        logging.info("OCT Segmentation pipeline started")
+
+        response = self.runSegmentation(vPAWOCTConfigFile)
+        if response:
+            model_path = self.getModelPathFromYAML(vPAWOCTConfigFile)
+            # Tell the user where to find the results
+            slicer.util.infoDisplay(
```

Comment:
This success message displays even when there was nothing to process. In the future it would probably be good to proclaim failure if there was nothing to process. But this will all be easier to do when OCTSeg has a proper inference workflow, so I think no need now